### ISSUE TITLE
Access storage from Android/iOS native

### ISSUE THREAD

**Issue by anirudhagarwal1** [ammarahm-ed/react-native-mmkv-storage#217]
Wanted to check how can I access the default storage instance to be read/write from the native Android/iOS code. 

This is required because I am working on a brownfield app and some of the values are updated/set in RN and need to be read in the native activities and vice versa. 

I was debugging the flow and found that the mmapId used we when instantiate MMKV storage from RN, it is "mmkvIDStore". But when I try to fetch MMKV storage object, it does not seem to hold the values set in RN. Correct me if I am wrong.

**Comment by ammarahm-ed** [comment]
Is the storage instance you are trying to access encrypted or not?

**Comment by ammarahm-ed** [comment]
> Is the storage instance you are trying to access encrypted or not? 

Also `mmkvIdStore` is an internal instance used by the library to hold some information regarding all instances that have been created.  It will not hold values you set from RN.

**Comment by anirudhagarwal1** [comment]
Not encrypted.

**Comment by anirudhagarwal1** [comment]
Can you help me with the mmapId I should use to get the instance used by RN? Is mmapId same as instance id?

**Comment by ammarahm-ed** [comment]
What is your initialization function for init the database in RN? Can you share it here??

**Comment by anirudhagarwal1** [comment]
In RN - 

```
const storage = new MMKVStorage.Loader().initialize();
storage.setInt('tempInt', 100);
```

**Comment by ammarahm-ed** [comment]
mmapID is same as instance id. You should use "default" as mmapID which should return the correct instance.

**Comment by anirudhagarwal1** [comment]
I did. Got the same instance with default. But got the incorrect value. 

Writing this on Android - 

```
MMKV instance = MMKV.mmkvWithID("default");
instance.containsKey("tempInt"); // returns true
instance.decodeInt("tempInt"); // return 0
```

**Comment by ammarahm-ed** [comment]
> I did. Got the same instance with default. But got the incorrect value. 
> 
> Writing this on Android - 
> 
> ```
> MMKV instance = MMKV.mmkvWithID("default");
> instance.containsKey("tempInt"); // returns true
> instance.decodeInt("tempInt"); // return 0
> ```
> 

That's because the value is stored as double.

**Comment by ammarahm-ed** [comment]
> > I did. Got the same instance with default. But got the incorrect value. 
> > 
> > Writing this on Android - 
> > 
> > ```
> > MMKV instance = MMKV.mmkvWithID("default");
> > instance.containsKey("tempInt"); // returns true
> > instance.decodeInt("tempInt"); // return 0
> > ```
> > 
> 

That's because the value is stored as double. 

> If containsKey returns true, it means you are accessing the correct instance.

**Comment by anirudhagarwal1** [comment]
> > I did. Got the same instance with default. But got the incorrect value.
> > Writing this on Android -
> > ```
> > MMKV instance = MMKV.mmkvWithID("default");
> > instance.containsKey("tempInt"); // returns true
> > instance.decodeInt("tempInt"); // return 0
> > ```
> 
> That's because the value is stored as double.

How can I retrieve it on the native side?

**Comment by ammarahm-ed** [comment]
Use `decodeDouble` method of mmkv.

**Comment by anirudhagarwal1** [comment]
> Use `decodeDouble` method of mmkv.

I am using 0.6.12 of the library and there seems to be no function like that. Am I missing something here?

**Comment by ammarahm-ed** [comment]
@anirudhagarwal1 I meant on Native Side.

```
MMKV instance = MMKV.mmkvWithID("default");
instance.containsKey("tempInt");
instance.decodeDouble("tempInt"); 

```

**Comment by anirudhagarwal1** [comment]
@ammarahm-ed I was talking about the native side itself. There seems to be no method called decodeDouble. 

Only method available are - 

<img width="519" alt="Screenshot 2022-02-12 at 7 30 59 PM" src="https://user-images.githubusercontent.com/85726789/153714411-679d5b37-1cad-48b3-80dd-ee177f2b2e8d.png">

There is no method to decode string as well.

**Comment by ammarahm-ed** [comment]
@anirudhagarwal1 Ah ok. The library implements only a subset of methods on Android and the decodeDouble methods is missing because it is not used. At the moment it won't be possible to do this unless these methods are added in .cpp file in JNI on android. 

This will require an update to be implemented.

**Comment by anirudhagarwal1** [comment]
Yep. Got it. Any plans to get those methods added in the near future?

**Comment by ammarahm-ed** [comment]
@anirudhagarwal1 I will add them in the next version

**Comment by anirudhagarwal1** [comment]
Awesome. Thanks @ammarahm-ed 
I would have loved to create a PR myself but have never worked on JNI myself. 
Any estimate on when the next version is coming out?

**Comment by arjun-anar** [comment]
any update on this ?

**Comment by arjun-anar** [comment]
@ammarahm-ed @anirudhagarwal1

**Comment by sidferreira** [comment]
@ammarahm-ed Does https://github.com/ammarahm-ed/react-native-mmkv-storage/issues/217#issuecomment-1037232765 still work? I'm having issues accessing MMKV on Activity/Java/AndroidStudio

<img width="632" alt="Screenshot 2024-01-15 at 2 12 16 PM" src="https://github.com/ammarahm-ed/react-native-mmkv-storage/assets/143615/b3d705be-a8d6-4e48-ac13-ed346b516a56">

**Comment by josephyanks** [comment]
@sidferreira That should work, how are you adding the dependency to your project? For #343 the way we tested used the same function set and worked properly:

```
MMKV.initialize(this)
val mmkv = MMKV.mmkvWithID("default")
Log.d("DEBUG_TAG", "Native mmkv contains key \"my_id\": ${mmkv?.containsKey("my_id")}")
```

**Comment by sidferreira** [comment]
@josephyanks I added that as a way to try to make it work... I dunno why it just shows that error messages on AndroidStudio...
